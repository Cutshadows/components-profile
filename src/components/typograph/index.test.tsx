import { render } from '@testing-library/react';
import Typograph from './index';

describe('Typograph', () => {
	it('renders the children correctly', () => {
		const { getByText } = render(
			<Typograph variant="p" applyDefaultStyles={true}>
				Hello World
			</Typograph>
		);
		expect(getByText('Hello World')).toBeInTheDocument();
	});

	it('applies the specified variant component', () => {
		const { container } = render(
			<Typograph variant="h1" applyDefaultStyles={true}>
				Heading 1
			</Typograph>
		);
		expect(container.firstChild).toContainElement(container.querySelector('h1'));
	});

	it('applies the specified custom className', () => {
		const { container } = render(
			<Typograph variant="p" className="custom-class" applyDefaultStyles={true}>
				Custom Class
			</Typograph>
		);
		expect(container.firstChild).toContainElement( container.querySelector('p'));
	});

	it('does not apply default styles when applyDefaultStyles is set to false', () => {
		const { container } = render(
			<Typograph variant="p" applyDefaultStyles={false}>
				No Default Styles
			</Typograph>
		);
		expect(container.firstChild).not.toHaveClass('default-styles');
	});


});