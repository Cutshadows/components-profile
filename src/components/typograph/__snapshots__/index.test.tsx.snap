// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Typograph > renders the correct text 1`] = `
<p
  class="_typography_a4eef9undefined"
>
  Hello World
</p>
`;
